fe/fe-core/src/main/java/com/starrocks/sql/analyzer/BackupRestoreAnalyzer.java

```diff
@@ -306,12 +308,19 @@ public static void analyzeTableRef(TableRef tableRef, String dbName, Database db
             }
         }
 
-        if (partitionNames != null) {
+        OlapTable olapTbl = null;
+        if (tbl instanceof OlapTable) {
+            olapTbl = (OlapTable) tbl;
+            PartitionInfo partitionInfo = olapTbl.getPartitionInfo();
+            if (partitionInfo instanceof ListPartitionInfo) {
+                throw new SemanticException("List partition table does not support backup/restore job");
+            }
+        }
+
+        if (olapTbl != null && partitionNames != null) {
```

Comment:
If olapTbl != null, !tbl.isOlapOrCloudNativeTable() is always false, it never hit the exception here. Maybe we should remove the olapTbl != null here?